src/SignalR/clients/ts/signalr/src/FetchHttpClient.ts

```diff
@@ -1,9 +1,36 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+// @ts-ignore: This will be removed from built files and is here to make the types available during dev work
+import * as tough from "@types/tough-cookie";
+
 import { AbortError, HttpError, TimeoutError } from "./Errors";
 import { HttpClient, HttpRequest, HttpResponse } from "./HttpClient";
 import { ILogger, LogLevel } from "./ILogger";
+import { Platform } from "./Utils";
+
+let abortControllerType: { prototype: AbortController, new(): AbortController };
+let fetchType: (input: RequestInfo, init?: RequestInit) => Promise<Response>;
+let jar: tough.CookieJar;
+if (typeof fetch === "undefined") {
+    // In order to ignore the dynamic require in webpack builds we need to do this magic
+    // @ts-ignore: TS doesn't know about these names
+    const requireFunc = typeof __webpack_require__ === "function" ? __non_webpack_require__ : require;
+
+    // Cookies aren't automatically handled in Node so we need to add a CookieJar to preserve cookies across requests
+    jar = new (requireFunc("tough-cookie")).CookieJar();
+    fetchType = requireFunc("node-fetch");
+
+    // node-fetch doesn't have a nice API for getting and setting cookies
+    // fetch-cookie will wrap a fetch implementation with a default CookieJar or a provided one
+    fetchType = requireFunc("fetch-cookie")(fetchType, jar);
+
+    // Node needs EventListener methods on AbortController which our custom polyfill doesn't provide
```

Comment:
How complex would it be to add it to our polyfill?

Reply:
Difficult because I have no idea how to work with the EventListener API.

Reply:
I guess we can't just use https://nodejs.org/api/events.html because our polyfill still has to work on the browser.